homeassistant/components/vallox/sensor.py

```diff
@@ -123,16 +107,10 @@ class ValloxSensorEntityDescription(SensorEntityDescription):
     """Describes Vallox sensor entity."""
 
     metric_key: str | None = None
-    sensor_type: type[ValloxSensor] = ValloxSensor
+    sensor_type: type[ValloxSensorEntity] = ValloxSensorEntity
 
 
-SENSORS: tuple[ValloxSensorEntityDescription, ...] = (
-    ValloxSensorEntityDescription(
-        key="current_profile",
-        name="Current profile",
-        icon="mdi:gauge",
-        sensor_type=ValloxProfileSensor,
-    ),
```

Comment:
What do you think, would it make sense to leave this in place for a release or two to avoid doing a breaking change at the same time as the replacement entity is added? OTOH, I don't know if there's any way of notifying users that the sensor will be removed in a future release and that the `select` entity should be used instead.

Reply:
"Type of change" topic description states that we should pick only one. So I am unable to make breaking changes in this release. =) That was my logic.

Reply:
Second this. We should give the user some time and hint to port over. That’s standard practice. For example, we did it with the fan service when porting over to the new fan entity as well.

Reply:
@andre-richter Do you know how to do this?

Reply:
One way I could think of is emitting a warning the first few (5, or 10) times native_value() for this sensor is called, hoping people discover those in the logs.

Compare to https://github.com/home-assistant/core/pull/66655 for what we did for the fan profile.

Reply:
I guess one option could be to just leave the existing `Current profile` sensor in place and don't block this PR on getting deprecation warnings in place? And then look into deprecations in a follow-up.

Reply:
One thing to consider, create the `Current profile` sensor as disabled by default if the Vallox integration is added by someone after this change:

```diff
diff --git homeassistant/components/vallox/sensor.py homeassistant/components/vallox/sensor.py
index 4fefed45fc..42398af910 100644
--- homeassistant/components/vallox/sensor.py
+++ homeassistant/components/vallox/sensor.py
@@ -64,6 +64,8 @@ class ValloxSensorEntity(ValloxEntity, SensorEntity):
 class ValloxProfileSensor(ValloxSensorEntity):
     """Child class for profile reporting."""
 
+    _attr_entity_registry_enabled_default = False
+
     @property
     def native_value(self) -> StateType:
         """Return the value reported by the sensor."""
```

Reply:
https://github.com/home-assistant/frontend/pull/13192 👀

No idea what it does yet, but it looks like a new HA feature that might help us with this.

Reply:
I'm still of the opinion that we should keep this sensor for a few months to avoid breaking things in the next release. There's no pressing need to immediately drop it. IMO, it's just a small annoyance that there are two sensors (temporarily) exposing the same data.

Reply:
I have returned it. But I think it also needs to log a deprecation message somehow if it is enabled.

Reply:
Yeah, I agree that a warning would be optimal. Maybe we'll know more about what https://github.com/home-assistant/frontend/pull/13192 is about once the next beta drops on Wednesday.

Reply:
You can also drop the "breaking change" part in the PR description 🤞